atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -775,6 +779,51 @@ public void deletesGetBatched() {
                 .isEqualTo(maxTsForFinePartition(0));
     }
 
+    @Test
+    public void multipleSweepersSweepDifferentShardsAndCallUnlockAfterwards() throws InterruptedException {
+        int shards = 128;
+        int sweepers = 8;
+        int threads = shards / sweepers;
+        LockService stickyLockService = createStickyLockService();
+        createAndInitializeSweepersAndWaitForOneBackgroundIteration(sweepers, shards, threads, stickyLockService);
+
+        for (int i = 0; i < shards; i++) {
+            assertProgressUpdatedToTimestamp(maxTsForFinePartition(tsPartitionFine(unreadableTs - 1)), i);
+            verify(stickyLockService, times(1)).unlock(new LockRefreshToken(BigInteger.valueOf(i), 1L));
+        }
+    }
+
+    @Test
+    public void extraSweepersGiveUpAfterFailingToAcquireEnoughTimes() throws InterruptedException {
+        int shards = 8;
+        int sweepers = 2;
+        int threads = shards / sweepers + 1;
+        LockService stickyLockService = createStickyLockService();
+        createAndInitializeSweepersAndWaitForOneBackgroundIteration(sweepers, shards, threads, stickyLockService);
+
+        // We issue between 2 * shards - shards/sweepers and 2 * shards - 1 lock requests to acquire all locks, and then
+        // an additional 2 * shards requests by the unlucky threads that have been unable to get a lock
+        ArgumentCaptor<LockRequest> captor = ArgumentCaptor.forClass(LockRequest.class);
```

Comment:
I'm not sure these bounds are correct generally, assuming you take `sweepers` as a variable (they are correct in this instance where `sweepers = 2`, but in that case I'd expect to see `2 * shards - shards / 2`).

More generally I get, where `shards / sweepers` is an integer, a minimum of `shards * (sweepers + 1) / 2` and a maximum of `shards * sweepers - (sweepers * (sweepers - 1)) / 2`

Reply:
@gmaretic I saw this test flake on circle with message "Wanted at most 314 times but was 322"

Reply:
Internal build #11157.